backend/modules/extract/services/ports/strategies.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+import logging
+
+
+logger = logging.getLogger(_name_)
+
+class ExtractStrategy(ABC):
```

Comment:
Please remember to add description to port class methods like this:
        """
        :param course_id: ID of course to retrieve.
        :param published: (Optional) Whether to retrieve only published course.
        :raises CourseDoesNotExist: Course with given ID doesn't exist.

        :return: Course output data for given ID.
        """
        
        
        and to adapters like this:
    """
    See description of parent class to get more details.
    """